src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs

```diff
@@ -71,7 +71,7 @@ public NumberBox()
 			Loaded += (s, e) => InitializeTemplate();
 			Unloaded += (s, e) => DisposeRegistrations();
 
-			DefaultStyleKey = this;
+			SetDefaultStyleKey(this);
```

Comment:
No it won't - this is the actual method: 
```csharp
		/// <summary>
		/// Duplicates the SetDefaultStyleKey() helper method from WinUI code.
		/// </summary>
		private protected void SetDefaultStyleKey<TDerived>(TDerived derivedControl) where TDerived : Control
			=> DefaultStyleKey = typeof(TDerived);
```

Since it's using the generic type parameter, it'll use the compile-time type, which is `NumberBox`.